ecs-cli/modules/cli/cluster/cluster_app.go

```diff
@@ -237,8 +239,32 @@ func createCluster(context *cli.Context, awsClients *AWSClients, commandConfig *
 		deleteStack = true
 	}
 
+	tags := make([]*ecs.Tag, 0)
+	if tagVal := context.String(flags.ResourceTagsFlag); tagVal != "" {
+		tags, err = utils.GetTags(tagVal, tags)
+		if err != nil {
+			return err
+		}
+	}
+
+	containerInstanceTaggingSupported := false
+
+	if len(tags) > 0 {
+		// determine if container instance tagging is supported
+		setting, err := getContainerInstanceARNSetting(awsClients.ECSClient)
+		if err != nil {
+			return err
+		}
+		if aws.StringValue(setting.Value) == ecsSettingEnabled {
+			containerInstanceTaggingSupported = true
+			logrus.Warnf("Enabling container instance tagging because %s is enabled for your identity, %s. If this is not your account default setting, your instances will fail to join your cluster. You can use the PutAccountSettingDefault API to change your account default.", ecs.SettingNameContainerInstanceLongArnFormat, aws.StringValue(setting.PrincipalArn))
```

Comment:
since this `Warnf` and the next one are the same except for the format string, can we set it as a var here and then execute it once so we don't need the `else` block? i.e.:
```go
warnMsgString := disableInstanceTagMsg   // store string in var

if aws.StringValue(setting.Value) == ecsSettingEnabled {
	containerInstanceTaggingSupported = true
        warnMsgString = enableInstanceTagMsg   // store in var
}

logrus.Warnf(warnMsgString, ecs.SettingNameContainerInstanceLongArnFormat, aws.StringValue(setting.PrincipalArn))

```

Reply:
Its not the same tho... the other one has the words "Disabling" vs "Enabling" and "disable" vs "enable".  I could put those into format strings... but it seems weird, so I'm gonna stick with the way it is.